src/python/pants/backend/python/rules/BUILD

```diff
@@ -46,9 +46,18 @@ python_tests(
   name='integration',
   sources=globs('*_integration_test.py'),
   dependencies=[
-    'src/python/pants/testutil:int-test',
-    'testprojects/tests/python/pants:dummies_directory',
+    ':rules',
+    'src/python/pants/backend/python/subsystems',
+    'src/python/pants/engine:fs',
+    'src/python/pants/engine:rules',
+    'src/python/pants/engine:selectors',
+    'src/python/pants/engine/legacy:structs',
+    'src/python/pants/rules/core',
+    'src/python/pants/source',
+    'src/python/pants/testutil:interpreter_selection_utils',
+    'src/python/pants/testutil:test_base',
+    'src/python/pants/testutil/subsystem',
   ],
   tags = {'integration', 'partially_type_checked'},
-  timeout = 120,
+  timeout = 480,
```

Comment:
The performance of these tests is atrocious, averaging 30 seconds per test. The main culprit is https://github.com/pantsbuild/pants/issues/8685. This bad performance is not permanent - it's been on my medium-term roadmap for a little bit now.